### PR TITLE
William Fischer independent challenge 2

### PR DESCRIPTION
**Pull request by wfischer42**

### CODE REVIEW

File: test/house_test.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  def test_it_exists
+    house = House.new("$400000", "123 sugar lane")
+    assert_instance_of House, house
+  end
+
+  def test_it_can_convert_string_price_to_integer
+    house = House.new("$400000", "123 sugar lane")
+    assert_equal 1231231, house.convert_price("$1231231")
+    assert_equal 1231231, house.convert_price("$1,231,231")
```
Comment (BrianZanti) [inline test/house_test.rb]: Great job following the spec! And you even through in another edge case!

File: test/house_test.rb [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  end
+
+  def test_it_is_initialized_with_parameters
+    house = House.new("$400000", "123 sugar lane")
+    assert_equal 400000, house.price
```
Comment (BrianZanti) [inline test/house_test.rb]: Great job following the spec!

File: test/house_test.rb (hunk 1 above)
Comment (BrianZanti) [inline test/house_test.rb]: Good job edge case testing!

File: test/house_test.rb [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  def test_it_has_no_rooms_by_default
+    house = House.new("$400000", "123 sugar lane")
+    assert_equal [], house.rooms
+  end
+
+  def test_rooms_can_be_added
+    house = House.new("$400000", "123 sugar lane")
+
+    assert_equal 0, house.rooms.size
```
Comment (BrianZanti) [inline test/house_test.rb]: This line isn't necessary. The previous test already shows us that a House has zero rooms by default

File: test/house_test.rb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    room_1 = Room.new(:bedroom, 10, 13)
+    room_2 = Room.new(:bedroom, 11, 15)
+
+    house.add_room(room_1)
+    house.add_room(room_2)
+
+    assert_equal room_1, house.rooms[0]
+    assert_equal room_2, house.rooms[1]
```
Comment (BrianZanti) [inline test/house_test.rb]: Instead of testing individually, test them both at once, i.e. `assert_equal [room_1, room_2], house.rooms`

File: test/house_test.rb [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    house.add_room(room_4)
+    house.add_room(room_1)
+    house.add_room(room_3)
+
+    rooms = house.rooms_by_category
+
+    assert rooms[:bedroom].include?(room_1)
+    assert rooms[:bedroom].include?(room_2)
+    assert rooms[:living_room].include?(room_3)
+    assert rooms[:basement].include?(room_4)
```
Comment (BrianZanti) [inline test/house_test.rb]: This is a common theme with your testing. Try to be as specific as possible with your expected values, i.e.

```ruby
expected = {
  bedroom: [room_1, room_1],
  living_room: [room_3],
  basement: [room_4]
}
```

File: lib/house.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    @rooms = []
+  end
+
+  def convert_price(price)
+    stripped_price = price.tr('$,', '')
+    stripped_price.to_i
+  end
+
+  def add_room(new_room)
+    if new_room.class == Room
```
Comment (BrianZanti) [inline lib/house.rb]: Interesting edge case guard! I would want to see a test for this behavior.

File: lib/house.rb [hunk 7]
```diff
@@ -0,0 +1,56 @@
+require './lib/room'
+
+class House
```
Comment (BrianZanti) [inline lib/house.rb]: Great job, Will!
